core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1755,6 +1765,78 @@ class ReplicaManager(val config: KafkaConfig,
     partitionsToMakeFollower
   }
 
+  private def updateTopicIdForFollowers(controllerId: Int,
```

Comment:
I talked to Jason about this and there were a few issues with the makeFollowers path. Namely `updateAssignmentAndIsr` in the `makeFollower` method. This can override a pending isr state and we do not want this.

Reply:
Right. Reusing `makeFollowers` would not work out of the box.

I am not sure to follow your point regarding the possibility of overriding a pending isr state. I thought that only the leader updates the isr state and the followers only updates it via `updateAssignmentAndIsr` based on the controller state. I might be missing something.

I am not against having `updateTopicIdForFollowers` but I wanted to ensure that we have thought about all the options.

Reply:
there is an issue with alter isr which is partially explained in this comment:
```
* With the addition of AlterIsr, we also consider newly added replicas as part of the ISR when advancing
* the HW. These replicas have not yet been committed to the ISR by the controller, so we could revert to the previously
* committed ISR. However, adding additional replicas to the ISR makes it more restrictive and therefor safe. We call
* this set the "maximal" ISR. See KIP-497 for more details
```

Reply:
I still don't see how a follower would use the alter isr api as a follower does not update the isr, only a leader does.